### PR TITLE
[Storage] `az storage account`: Add `az storage account local-user` and `--enable-sftp` and `--enable-local-user` to `az storage account create/update`

### PR DESCRIPTION
**Pull request by calvinhzy**
**Related command**

**Description**

Add `az storage account local-user` and `--enable-sftp` and `--enable-local-user` to `az storage account create/update` 
GA these commands/args which were in extension before. 

**Testing Guide**

`az storage account create -n {sa} -g {rg} -l {loc} --sku Standard_LRS --enable-sftp true --enable-local-user true`
`az storage account local-user create --account-name {sa} -g {rg} -n {username} --home-directory home '\
'--permission-scope permissions=r service=blob resource-name=container1 '\
'--permission-scope permissions=rw service=file resource-name=share2 '\
'--has-ssh-key false --has-shared-key false`

**History Notes**

[Storage] `az storage account local-user`: New command group to manage identities when using SFTP
[Storage] `az storage account create/update`: Add `--enable-sftp` and `--enable-local-user` to support SSH File Transfer Protocol

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -442,6 +447,10 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('use_subdomain', help='Specify whether to use indirect CNAME validation.',
                    arg_type=get_enum_type(['true', 'false']))
         c.argument('tags', tags_type, default=None)
+        c.argument('enable_sftp', arg_type=get_three_state_flag(), min_api='2021-08-01',
+                   is_preview=True, help='Enable Secure File Transfer Protocol.')
+        c.argument('enable_local_user', arg_type=get_three_state_flag(), min_api='2021-08-01',
+                   is_preview=True, help='Enable local user features.')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:453]:
```suggestion
        c.argument('enable_sftp', arg_type=get_three_state_flag(), min_api='2021-08-01',
                   help='Enable Secure File Transfer Protocol.')
        c.argument('enable_local_user', arg_type=get_three_state_flag(), min_api='2021-08-01',
                   help='Enable local user features.')
```
Service is requesting for GA SFTP

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 2]
```diff
@@ -764,6 +773,39 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                    arg_type=get_enum_type(['validation', 'upgrade']), validator=validate_hns_migration_type,
                    help='Start a validation request for migration or start a migration request')
 
+    with self.argument_context('storage account local-user') as c:
+        c.argument('account_name', acct_name_type, options_list='--account-name', id_part=None)
+        c.argument('username', options_list=['--username', '--name', '-n'],
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:778]:
```suggestion
        c.argument('username', options_list=['--user-name', '--name', '-n'],
```

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 3]
```diff
@@ -2222,3 +2222,45 @@ def validate_fs_file_set_expiry(namespace):
...
+            permissions, service, resource_name = '', '', ''
+            for s in values:
+                if "permissions" in s:
+                    permissions = s.split('=')[1]
+                elif "service" in s:
+                    service = s.split('=')[1]
+                elif "resource-name" in s:
+                    resource_name = s.split('=')[1]
+        except (ValueError, TypeError):
+            raise CLIError('usage error: --permission-scope VARIABLE OPERATOR VALUE')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py:2243]:
```suggestion
            for s in values:
                k, v = s.split('=', 1)
                if k == "permissions":
                    permissions = v
                elif k == "service":
                    service = v
                elif k == "resource-name":
                    resource_name = v
                else:
                    raise UnrecognizedArgumentError('key error: key must be one of permissions, service, resource-name for --permission-scope')
        except (ValueError, TypeError, IndexError):
            raise CLIError('usage error: --permission-scope [Key=Value ...]')
```

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 4]
```diff
@@ -2222,3 +2222,45 @@ def validate_fs_file_set_expiry(namespace):
...
+        SshPublicKey = namespace._cmd.get_models('SshPublicKey')
+        try:
+            description, key = '', ''
+            for k in values:
+                if "description" in k:
+                    description = k.split('=')[1]
+                elif "key" in k:
+                    key = k.split('=')[1]
+        except (ValueError, TypeError):
+            raise CLIError('usage error: --ssh-authorized-key VARIABLE OPERATOR VALUE')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py:2265]: Same problem with https://github.com/Azure/azure-cli/pull/23611/files#r952241516

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**